src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/RemoveBuildingUnitLambdaHandler.cs

```diff
@@ -0,0 +1,69 @@
+namespace BuildingRegistry.Api.BackOffice.Handlers.Lambda.Handlers.BuildingUnit
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Be.Vlaanderen.Basisregisters.AggregateSource;
+    using Be.Vlaanderen.Basisregisters.Sqs.Exceptions;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Handlers;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Infrastructure;
+    using Be.Vlaanderen.Basisregisters.Sqs.Responses;
+    using BuildingRegistry.Api.BackOffice.Abstractions.Building.Validators;
+    using BuildingRegistry.Building;
+    using BuildingRegistry.Building.Exceptions;
+    using Microsoft.Extensions.Configuration;
+    using Requests.BuildingUnit;
+    using TicketingService.Abstractions;
+
+    public sealed class RemoveBuildingUnitLambdaHandler : BuildingUnitLambdaHandler<RemoveBuildingUnitLambdaRequest>
+    {
+        public RemoveBuildingUnitLambdaHandler(
+            IConfiguration configuration,
+            ICustomRetryPolicy retryPolicy,
+            ITicketing ticketing,
+            IIdempotentCommandHandler idempotentCommandHandler,
+            IBuildings buildings)
+            : base(
+                configuration,
+                retryPolicy,
+                ticketing,
+                idempotentCommandHandler,
+                buildings)
+        { }
+
+        protected override async Task<ETagResponse> InnerHandle(RemoveBuildingUnitLambdaRequest request, CancellationToken cancellationToken)
+        {
+            var cmd = request.ToCommand();
+
+            try
+            {
+                await IdempotentCommandHandler.Dispatch(
+                    cmd.CreateCommandId(),
+                    cmd,
+                    request.Metadata,
+                    cancellationToken);
+            }
+            catch (IdempotencyException)
+            {
+                // Idempotent: Do Nothing return last etag
+            }
+
+            var lastHash = await GetHash(
+                request.BuildingPersistentLocalId,
+                new BuildingUnitPersistentLocalId(request.BuildingUnitPersistentLocalId),
+                cancellationToken);
+
+            return new ETagResponse(string.Format(DetailUrlFormat, request.BuildingUnitPersistentLocalId), lastHash);
```

Comment:
@ArneD should we include lastHash in etag response or pass empty string?

Reply:
Last hash is good (consistent with address).